microsoft/testsuites/core/azure_image_standard.py

```diff
@@ -894,6 +905,48 @@ def verify_boot_error_fail_warnings(self, node: Node) -> None:
             f" {node.os.name} {node.os.information.version}"
         ).is_empty()
 
+    @TestCaseMetadata(
+        description="""
+        This test will check ERROR, WARNING messages from /var/log/cloud-init.log
+        and also check cloud-init exit status.
+
+        Steps:
+        1. Get ERROR, WARNING messages from /var/log/cloud-init.log.
+        2. If any unexpected ERROR, WARNING messages or non-zero cloud-init status
+         fail the case.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_platform_type=[AZURE, READY]),
+    )
+    def verify_cloud_init_error_status(self, node: Node) -> None:
+        cat = node.tools[Cat]
+        if isinstance(node.os, CBLMariner):
+            if node.shell.exists(node.get_pure_path("/var/log/cloud-init.log")):
+                log_output = cat.read("/var/log/syslog", force_run=True, sudo=True)
+
+                found_results = [
+                    x
+                    for sublist in find_patterns_in_lines(
+                        log_output, self._ERROR_WARNING_pattern
+                    )
+                    for x in sublist
+                    if x
+                ]
+                assert_that(found_results).described_as(
+                    "unexpected ERROR/WARNING shown up in cloud-init.log"
+                    f" {node.os.name} {node.os.information.version}"
+                ).is_empty()
+                cmd_result = node.execute("cloud-init status --wait", sudo=True)
+                if 0 != cmd_result.exit_code:
+                    raise LisaException(
+                        "cloud-init status failed with exit_code"
+                        f" {cmd_result.exit_code}."
+                    )
+            else:
+                raise LisaException("cloud-init.log not exists")
+        else:
+            return
```

Comment:
If it's other images, it should skip instead of pass.

Reply:
change as suggested

Reply:
Please skip for other images by raise a skipped exception. If nothing to do, the test case will be passed, it's confusing.